### PR TITLE
Unit tests for greaterThan, greaterThanEqual, lessThanEqual, & matmul

### PR DESCRIPTION
**Pull request by cryptodeal**
Now that we've fixed the bug in `buffer`, wanted to get the unit tests for greaterThan, greaterThanEqual, & lessThanEqual added.

### Supported Operations Tests
- [x] abs (TODO: addtl tests in `abs.test.ts` after basic op coverage)
- [x] absolute (implicit, since the same as `abs`)
- [x] add
- [x] all
- [ ] amax
- [ ] amin
- [ ] any
- [ ] arange
- [x] argmax
- [x] argmin
- [ ] backward
- [ ] bitwiseAnd
- [ ] bitwiseOr
- [ ] bitwiseXor
- [ ] bytesUsed
- [x] ceil
- [ ] clip
- [x] cos
- [ ] countNonzero
- [ ] cumsum
- [x] div
- [ ] eq
- [x] erf
- [x] exp
- [ ] flip (WIP; failing)
- [x] floor
- [ ] full
- [x] greaterThan
- [x] greaterThanEqual
- [ ] identity
- [ ] iota
- [x] isinf
- [x] isnan
- [ ] lShift
- [x] lessThan
- [x] lessThanEqual
- [x] log
- [ ] log1p
- [ ] logicalAnd
- [ ] logicalNot
- [ ] logicalOr
- [x] matmul
- [x] maximum
- [x] mean
- [ ] median
- [x] minimum
- [ ] mod
- [x] mul
- [x] negative
- [ ] neq
- [ ] nonzero
- [x] norm
- [ ] power
- [ ] rShift
- [ ] rand
- [ ] randn
- [x] reshape
- [ ] rint
- [ ] roll
- [x] scalar
- [x] sigmoid
- [x] sign
- [x] sin
- [ ] sort
- [ ] sqrt
- [ ] std
- [x] sub
- [x] sum
- [x] tanh
- [x] tensor
- [x] tile
- [x] transpose
- [ ] tril
- [ ] triu
- [ ] where

### Tensor Class Methods & Properties Tests
- [ ] asContiguousTensor
- [x] elements (used in tests)
- [ ] ndim
- [ ] ptr (need test for this?)
- [x] shape (used in tests)
- [ ] shape64
- [x] copy
- [ ] deepCopy
- [ ] detach
- [x] index
- [x] indexedAssign
- [ ] requireGrad
- [x] toFloat32
- [x] toFloat32Array
- [ ] toString
- [ ] update
- [ ] valueOf
- [ ] var

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
       "@shumai/shumai": ["./shumai/index.ts"],
     }
   },
-  "exclude": ["examples/**/*", "build/**/*", "docs/assets/**/*"],
+  "exclude": ["examples/**/*.js", "build/**/*", "docs/assets/**/*"],
```
Comment (bwasti) [inline tsconfig.json:18]: if this is `.js` it breaks the docs for me -- mind changing it back for the time being?

Comment (cryptodeal) [reply tsconfig.json:18]: No problem! Give me a few as I'm updating the bulleted list in the OG Pr comment to match the order operations are listed in the docs.

Comment (cryptodeal) [reply tsconfig.json]: Done!

**Comment by cryptodeal** [comment]
A few tests fail as the actual shape does not equal expected shape, so added logged output of actual vs expected shape only when failing `isShape`.

**Pull request merged**